Add workflow toggle for environments without Azure OIDC permissions

## Problem Description

GitHub Actions workflows are failing due to Azure OIDC authentication errors when the Azure service principal does not have the required role assignments. This occurs when users lack permissions to assign roles during the `scripts/setup-backend.sh` execution.

### Current Failure

```
##[error]No subscriptions found for ***.
##[error]Login failed with Error: The process '/usr/bin/az' failed with exit code 1
```

**Root Cause**: The Azure service principal is created but lacks:
- `Contributor` role at subscription level
- `Storage Blob Data Owner` role on storage account

Without these roles, OIDC authentication in GitHub Actions fails, causing all Terraform workflows to fail.

## Current Workflow Dependencies

### Workflows Requiring Azure OIDC (Currently Failing)
1. **terraform-init.yml** - Azure Login via OIDC for state backend access
2. **terraform-plan.yml** - Calls init + requires OIDC for plan execution
3. **terraform-apply.yml** - Calls init + requires OIDC for apply execution

### Workflows NOT Requiring Azure (Working)
1. **terraform-validate.yml** - Syntax/format validation only
2. **branch-cleanup.yml** - Git operations only

## Existing Detection in setup-backend.sh

The setup script **already tracks** role assignment status:

```bash
# Lines 338-384 in scripts/setup-backend.sh
CONTRIBUTOR_ASSIGNED=false
STORAGE_ROLE_ASSIGNED=false
ROLES_ASSIGNED=false

# Assign Contributor role (lines 348-360)
if az role assignment create ...; then
  CONTRIBUTOR_ASSIGNED=true
else
  CONTRIBUTOR_ASSIGNED=false
fi

# Assign Storage Blob Data Owner (lines 367-378)
if az role assignment create ...; then
  STORAGE_ROLE_ASSIGNED=true
else
  STORAGE_ROLE_ASSIGNED=false
fi

# Both roles must succeed (lines 382-384)
if [ "$CONTRIBUTOR_ASSIGNED" = true ] && [ "$STORAGE_ROLE_ASSIGNED" = true ]; then
  ROLES_ASSIGNED=true
fi
```

## Proposed Solution

### 1. Use GitHub Repository Variable for Toggle

**Variable Name**: `ENABLE_AZURE_WORKFLOWS`

**Values**:
- `true` - Enable all Azure-dependent workflows (default when not set)
- `false` - Disable Azure-dependent workflows, keep validation only

**Why GitHub Variables (not Secrets)**:
- ✅ Visible in repository settings UI
- ✅ Can be easily toggled by users
- ✅ Not encrypted (appropriate for non-sensitive boolean)
- ✅ Accessible via `vars.ENABLE_AZURE_WORKFLOWS`

### 2. Modify Workflows with Conditionals

Add `if:` conditions to jobs requiring Azure OIDC:

#### terraform-plan.yml
```yaml
jobs:
  terraform-init:
    name: Initialize
    if: vars.ENABLE_AZURE_WORKFLOWS \!= 'false'  # Skip if disabled
    uses: ./.github/workflows/terraform-init.yml
    # ... rest of job

  terraform-plan:
    name: Plan Infrastructure Changes
    if: vars.ENABLE_AZURE_WORKFLOWS \!= 'false'  # Skip if disabled
    needs: terraform-validate
    # ... rest of job
```

#### terraform-apply.yml
```yaml
jobs:
  terraform-init:
    name: Initialize
    if: vars.ENABLE_AZURE_WORKFLOWS \!= 'false'  # Skip if disabled
    # ... rest of job

  terraform-plan:
    name: Plan Changes
    if: vars.ENABLE_AZURE_WORKFLOWS \!= 'false'  # Skip if disabled
    # ... rest of job

  terraform-apply:
    name: Apply Infrastructure Changes
    if: vars.ENABLE_AZURE_WORKFLOWS \!= 'false'  # Skip if disabled
    # ... rest of job
```

#### terraform-init.yml
```yaml
jobs:
  terraform-init:
    name: Initialize Terraform
    if: vars.ENABLE_AZURE_WORKFLOWS \!= 'false'  # Skip if disabled
    runs-on: ubuntu-latest
    # ... rest of job
```

**Note**: `terraform-validate.yml` requires **NO changes** - it doesn't use Azure OIDC.

### 3. Update setup-backend.sh

Add automatic variable configuration based on role assignment status:

```bash
# After line 526 (end of secrets creation logic)

# Configure GitHub workflow toggle based on role assignment status
if [ -n "$APP_ID" ] && [ "$ROLES_ASSIGNED" = true ]; then
  # Roles successfully assigned - enable Azure workflows
  if command -v gh &>/dev/null && gh auth status &>/dev/null; then
    echo "true" | gh variable set ENABLE_AZURE_WORKFLOWS --repo "$GITHUB_ORG/$GITHUB_REPO" 2>/dev/null || true
    print_success "Enabled Azure workflows (ENABLE_AZURE_WORKFLOWS=true)"
  fi
elif [ -n "$APP_ID" ] && [ "$ROLES_ASSIGNED" = false ]; then
  # Service principal exists but roles not assigned - disable Azure workflows
  if command -v gh &>/dev/null && gh auth status &>/dev/null; then
    echo "false" | gh variable set ENABLE_AZURE_WORKFLOWS --repo "$GITHUB_ORG/$GITHUB_REPO" 2>/dev/null || true
    print_warning "Disabled Azure workflows due to missing role assignments"
    print_info "To enable later, assign roles and run: gh variable set ENABLE_AZURE_WORKFLOWS --body 'true'"
  fi
fi
```

### 4. Documentation Updates

Add to setup script output summary (around line 556):

```bash
if [ "$ROLES_ASSIGNED" = false ]; then
  echo "  ⚠️  GitHub Actions workflows using Azure OIDC are DISABLED"
  echo "     Reason: Service principal lacks required role assignments"
  echo "     Only validation workflows will run"
  echo ""
  echo "  To enable Azure workflows after assigning roles:"
  echo "     gh variable set ENABLE_AZURE_WORKFLOWS --body 'true' --repo $GITHUB_ORG/$GITHUB_REPO"
fi
```

## Benefits of This Approach

1. ✅ **Non-destructive** - Workflows remain in repository, just skip when disabled
2. ✅ **Automatic detection** - setup-backend.sh sets variable based on permissions
3. ✅ **Manual override** - Users can change variable anytime via GitHub UI or CLI
4. ✅ **Backward compatible** - Default behavior (variable not set) = workflows run
5. ✅ **Graceful degradation** - Validation still runs, only Azure-dependent jobs skip
6. ✅ **Clear messaging** - Skipped jobs show why they didn't run
7. ✅ **Easy re-enable** - Once roles assigned, just update variable to `true`

## Testing Plan

### Test Case 1: Disable Workflows
```bash
# Set variable to false
gh variable set ENABLE_AZURE_WORKFLOWS --body 'false' --repo robinmordasiewicz/f5-xc-ce-terraform

# Trigger workflow (create PR touching terraform/)
# Expected: terraform-init and terraform-plan jobs skip, terraform-validate runs
```

### Test Case 2: Enable Workflows
```bash
# Set variable to true (after roles assigned)
gh variable set ENABLE_AZURE_WORKFLOWS --body 'true' --repo robinmordasiewicz/f5-xc-ce-terraform

# Trigger workflow
# Expected: All jobs run normally
```

### Test Case 3: Variable Not Set (Default)
```bash
# Delete variable
gh variable delete ENABLE_AZURE_WORKFLOWS --repo robinmordasiewicz/f5-xc-ce-terraform

# Trigger workflow
# Expected: All jobs run (backward compatible behavior)
```

## Implementation Checklist

- [ ] Update `.github/workflows/terraform-init.yml` with conditional
- [ ] Update `.github/workflows/terraform-plan.yml` with conditional
- [ ] Update `.github/workflows/terraform-apply.yml` with conditional
- [ ] Update `scripts/setup-backend.sh` to set variable based on ROLES_ASSIGNED
- [ ] Update setup script output messaging
- [ ] Test with variable set to `false`
- [ ] Test with variable set to `true`
- [ ] Test with variable not set (default behavior)
- [ ] Document in README how to manually toggle workflows

## Related Issues

- Related to #60 - Backend configuration (now resolved)
- Related to #59 - Service principal role assignment failures

## Manual Commands Reference

```bash
# View current variable value
gh variable list --repo robinmordasiewicz/f5-xc-ce-terraform

# Disable Azure workflows
gh variable set ENABLE_AZURE_WORKFLOWS --body 'false' --repo robinmordasiewicz/f5-xc-ce-terraform

# Enable Azure workflows
gh variable set ENABLE_AZURE_WORKFLOWS --body 'true' --repo robinmordasiewicz/f5-xc-ce-terraform

# Delete variable (revert to default behavior)
gh variable delete ENABLE_AZURE_WORKFLOWS --repo robinmordasiewicz/f5-xc-ce-terraform
```

--- a/.github/workflows/terraform-apply.yml
+++ b/.github/workflows/terraform-apply.yml
@@ -16,6 +16,8 @@ permissions:
 jobs:
   terraform-init:
     name: Initialize
+    # Skip if ENABLE_AZURE_WORKFLOWS is explicitly set to 'false'
+    if: vars.ENABLE_AZURE_WORKFLOWS != 'false'
     uses: ./.github/workflows/terraform-init.yml
     secrets:
       AZURE_CLIENT_ID: ${{ secrets.AZURE_CLIENT_ID }}
@@ -33,6 +35,8 @@ jobs:
 
   terraform-plan:
     name: Plan Changes
+    # Skip if ENABLE_AZURE_WORKFLOWS is explicitly set to 'false'
+    if: vars.ENABLE_AZURE_WORKFLOWS != 'false'
     needs: terraform-validate
     runs-on: ubuntu-latest
 
@@ -96,7 +100,8 @@ jobs:
   terraform-apply:
     name: Apply Infrastructure Changes
     needs: terraform-plan
-    if: needs.terraform-plan.outputs.has_changes == 'true'
+    # Skip if ENABLE_AZURE_WORKFLOWS is explicitly set to 'false' OR if no changes detected
+    if: vars.ENABLE_AZURE_WORKFLOWS != 'false' && needs.terraform-plan.outputs.has_changes == 'true'
     runs-on: ubuntu-latest
 
     environment:

--- a/.github/workflows/terraform-init.yml
+++ b/.github/workflows/terraform-init.yml
@@ -32,6 +32,9 @@ on:
 jobs:
   terraform-init:
     name: Initialize Terraform
+    # Skip if ENABLE_AZURE_WORKFLOWS is explicitly set to 'false'
+    # Default behavior (variable not set) is to run the workflow
+    if: vars.ENABLE_AZURE_WORKFLOWS != 'false'
     runs-on: ubuntu-latest
 
     permissions:

--- a/.github/workflows/terraform-plan.yml
+++ b/.github/workflows/terraform-plan.yml
@@ -16,6 +16,8 @@ permissions:
 jobs:
   terraform-init:
     name: Initialize
+    # Skip if ENABLE_AZURE_WORKFLOWS is explicitly set to 'false'
+    if: vars.ENABLE_AZURE_WORKFLOWS != 'false'
     uses: ./.github/workflows/terraform-init.yml
     secrets:
       AZURE_CLIENT_ID: ${{ secrets.AZURE_CLIENT_ID }}
@@ -33,6 +35,8 @@ jobs:
 
   terraform-plan:
     name: Plan Infrastructure Changes
+    # Skip if ENABLE_AZURE_WORKFLOWS is explicitly set to 'false'
+    if: vars.ENABLE_AZURE_WORKFLOWS != 'false'
     needs: terraform-validate
     runs-on: ubuntu-latest
 

--- a/scripts/setup-backend.sh
+++ b/scripts/setup-backend.sh
@@ -525,6 +525,48 @@ else
   SECRETS_SKIPPED=true
 fi
 
+# Configure GitHub workflow toggle based on role assignment status
+# This variable controls whether Azure-dependent workflows (init, plan, apply) run
+echo ""
+print_info "Configuring GitHub Actions workflow toggle..."
+
+if command -v gh &>/dev/null && gh auth status &>/dev/null 2>&1; then
+  if [ -n "$APP_ID" ] && [ "$ROLES_ASSIGNED" = true ]; then
+    # Roles successfully assigned - enable Azure workflows
+    if echo "true" | gh variable set ENABLE_AZURE_WORKFLOWS --repo "$GITHUB_ORG/$GITHUB_REPO" 2>/dev/null; then
+      print_success "Enabled Azure workflows (ENABLE_AZURE_WORKFLOWS=true)"
+      print_info "GitHub Actions will run terraform init/plan/apply with OIDC authentication"
+    else
+      print_warning "Could not set ENABLE_AZURE_WORKFLOWS variable automatically"
+      print_info "Set manually: gh variable set ENABLE_AZURE_WORKFLOWS --body 'true' --repo $GITHUB_ORG/$GITHUB_REPO"
+    fi
+  elif [ -n "$APP_ID" ] && [ "$ROLES_ASSIGNED" = false ]; then
+    # Service principal exists but roles not assigned - disable Azure workflows
+    if echo "false" | gh variable set ENABLE_AZURE_WORKFLOWS --repo "$GITHUB_ORG/$GITHUB_REPO" 2>/dev/null; then
+      print_warning "Disabled Azure workflows (ENABLE_AZURE_WORKFLOWS=false)"
+      print_info "Reason: Service principal lacks required role assignments"
+      print_info "Only terraform-validate will run in GitHub Actions"
+      echo ""
+      print_info "To enable Azure workflows after assigning roles:"
+      print_info "  gh variable set ENABLE_AZURE_WORKFLOWS --body 'true' --repo $GITHUB_ORG/$GITHUB_REPO"
+    else
+      print_warning "Could not set ENABLE_AZURE_WORKFLOWS variable automatically"
+      print_info "Set manually: gh variable set ENABLE_AZURE_WORKFLOWS --body 'false' --repo $GITHUB_ORG/$GITHUB_REPO"
+    fi
+  else
+    # No service principal created
+    print_info "No service principal - skipping workflow toggle configuration"
+  fi
+else
+  print_warning "GitHub CLI not available or not authenticated - skipping workflow toggle"
+  print_info "You can manually set the variable later:"
+  if [ "$ROLES_ASSIGNED" = true ]; then
+    print_info "  gh variable set ENABLE_AZURE_WORKFLOWS --body 'true' --repo $GITHUB_ORG/$GITHUB_REPO"
+  else
+    print_info "  gh variable set ENABLE_AZURE_WORKFLOWS --body 'false' --repo $GITHUB_ORG/$GITHUB_REPO"
+  fi
+fi
+
 # Summary
 echo ""
 print_success "✨ Azure backend setup complete!"
@@ -548,6 +590,10 @@ if [ "$SECRETS_CREATED" = true ] && [ "$SECRETS_SKIPPED" = false ]; then
   echo "   • ARM_RESOURCE_GROUP_NAME, ARM_STORAGE_ACCOUNT_NAME"
   echo "   • ARM_CONTAINER_NAME, ARM_KEY"
   echo ""
+  echo "✅ GitHub Actions Workflows:"
+  echo "   • ENABLE_AZURE_WORKFLOWS=true (workflows enabled)"
+  echo "   • terraform-init, terraform-plan, terraform-apply will run with OIDC"
+  echo ""
   echo "Next Steps:"
   echo "  1. Get F5 XC API token and add as F5_XC_API_TOKEN secret:"
   echo "     echo \"\$F5_XC_API_TOKEN\" | gh secret set F5_XC_API_TOKEN --repo $GITHUB_ORG/$GITHUB_REPO"
@@ -569,6 +615,15 @@ else
     echo "  AZURE_CLIENT_ID=\"$APP_ID\""
   fi
   echo ""
+  if [ "$ROLES_ASSIGNED" = false ]; then
+    echo "⚠️  GitHub Actions Workflows:"
+    echo "   • ENABLE_AZURE_WORKFLOWS=false (Azure workflows disabled)"
+    echo "   • Reason: Service principal lacks required role assignments"
+    echo "   • Only terraform-validate will run in GitHub Actions"
+    echo "   • To enable after assigning roles:"
+    echo "     gh variable set ENABLE_AZURE_WORKFLOWS --body 'true' --repo $GITHUB_ORG/$GITHUB_REPO"
+    echo ""
+  fi
   echo "Next Steps:"
 
   # Check if backend.local.hcl was generated

